aten/src/TH/generic/THTensorMath.cpp

```diff
@@ -298,20 +298,44 @@ static void THTensor_(addmmImpl)(THTensor *r_, THTensor *t, THTensor *m1, THTens
   int64_t ldm1_ = (transpose_m1 == 'n' ? m1_->stride((transpose_r == 'n' ? 1 : 0)) : m1_->stride((transpose_r == 'n' ? 0 : 1)));
   int64_t ldm2_ = (transpose_m2 == 'n' ? m2_->stride((transpose_r == 'n' ? 1 : 0)) : m2_->stride((transpose_r == 'n' ? 0 : 1)));
 
-  /* do the operation */
-  THBlas_(gemm)(transpose_m1,
-                transpose_m2,
-                m,
-                n,
-                k,
-                alpha,
-                m1_->data<scalar_t>(),
-                ldm1_,
-                m2_->data<scalar_t>(),
-                ldm2_,
-                beta,
-                r__->data<scalar_t>(),
-                ldr__);
+  // Don't go through GEMM if result is empty matrix, since this is not
+  // supported by BLAS.
+  if (m != 0 && n != 0) {
```

Comment:
I'm a bit confused... If the result is empty, what are these assignments doing here?

Reply:
There're 2 main cases when failure in BLAS routines happens:

1. n or m is 0, which will result in the final matrix that 0 as one of the dimensions (as an example Linear called on a batch of size 0). In this case the only part that is needed from this function - resize of the final matrix to have correct output dimensions.

2. Another case when BLAS can fail is when k is 0, that can fail when one of the strides is 0 (happens on backward pass for Linear with an empty batch). In that case I'm getting rid of no-op part for ADDMM:
alpha * A x B + beta * C -> beta * C